creatures/creatures.cpp

```diff
@@ -930,13 +981,459 @@ int Player::save(bool updateTime, LoadType saveType) {
     return(0);
 }
 
+//********************************************************************************************************
+//                                       checkResistEnchantments
+//********************************************************************************************************
+// This function can be called to check against any racial/class/other resistances to various enchantment
+// and hold spells and return true if the creature resists. Send false for output if do not want to print 
+// anything and are just checking resistances in any various logic
+
+bool Creature::checkResistEnchantments(const std::shared_ptr<Creature>& caster, const std::string spell, bool print) {
+    bool holdUndead = (spell == "hold-undead"), holdAnimal = (spell == "hold-animal"), 
+         holdPlant = (spell == "hold-plant"), holdPerson = (spell == "hold-person"), 
+         holdMonster = (spell == "hold-monster"), holdElemental = (spell == "hold-elemental"),
+         holdFey = (spell == "hold-fey");
+    bool resist=false, aggro=false, monsterImmune=false;
+    bool output = print;
+
+    if (!caster)
+        return(false);
+
+    if(isStaff())
+        return(true);
+
+    //if(caster->isStaff())
+     //   return(false);
```

Comment:
```suggestion
```

Reply:
Changed this to a TODO comment for after p-testing is over for hold spells.